dumper/app/src/test/java/com/google/edwmigration/dumper/application/dumper/connector/teradata/TeradataSqlQueryFactoryTest.java

```diff
@@ -0,0 +1,77 @@
+package com.google.edwmigration.dumper.application.dumper.connector.teradata;
+
+import static org.junit.Assert.assertEquals;
+
+import com.google.edwmigration.dumper.application.dumper.connector.ZonedInterval;
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.Arrays;
+import java.util.Collections;
+import org.junit.Test;
+
+public class TeradataSqlQueryFactoryTest {
+
+  @Test
+  public void testGetSql() {
+    //  Arrange
+    ZonedDateTime endDate = ZonedDateTime.parse("2023-03-09T00:00:00Z");
+    ZonedDateTime startDate = endDate.minusDays(5).truncatedTo(ChronoUnit.DAYS);
+    ZonedInterval interval = new ZonedInterval(startDate, endDate);
+    TeradataSqlQueryFactory factory =
+        new TeradataSqlQueryFactory(
+            interval,
+            "LT",
```

Comment:
nit: I would suggest leaving a comment before string parameters for cases where it is not obvious what the parameter stands for. The comment will be equal to the parameter name, e.g. `/* logTable= */ "LT",`. As an alternative, you can extract them to variables or give more obvious names, such as `testLogTable`